playground/frontend/lib/components/split_view/split_view.dart

```diff
@@ -0,0 +1,175 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import 'package:flutter/material.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:playground/config/theme.dart';
+import 'package:playground/constants/assets.dart';
+
+enum SplitViewDirection {
+  vertical,
+  horizontal,
+}
+
+const minRatio = 0.3;
+const maxRatio = 0.7;
+const defaultRatio = 0.5;
+
+class SplitView extends StatefulWidget {
+  final Widget first;
+  final Widget second;
+  final double dividerSize;
+  final SplitViewDirection direction;
+  final double ratio;
+
+  const SplitView({
+    Key? key,
+    required this.first,
+    required this.second,
+    required this.dividerSize,
+    required this.direction,
+    this.ratio = defaultRatio,
+  }) : super(key: key);
+
+  @override
+  _SplitViewState createState() => _SplitViewState();
+}
+
+class _SplitViewState extends State<SplitView> {
+  //from minRatio to maxRatio
+  double _ratio = defaultRatio;
+  double? _maxSize;
```

Comment:
Using a nullable type here seem like a code smell. Perhaps initialize the _maxSize to 0 and then remove the exclamation marks. The exclamation marks are most likely converting the value to a zero valued non-nullable int anyway.